Added logic to expand environment variables in key file path

KuttKatrea opened the pull request:
Environment variables were expanded just for the cygwin socket file, so I took that logic and added it to the key file loading, so we can use also Environment Variables in external key files.

Resolves #148

dlech commented:
Since we are using the same function, can we refactor to use a common function instead of copying it?

You could turn it into an extension method and put it in https://github.com/dlech/KeeAgent/blob/master/KeeAgent/ExtensionMethods.cs

KuttKatrea commented:
I have refactored the function, and updated the code to use it. I leave the `ResolvedFilename` property in `EntrySettings` because is easier to remember to use it instead of doing `ExtensionMethod.ExpandVariables(entrySettings.FileName)` each time.

Also, rebased on top of master.

dlech commented on KeeAgent/ExtensionMethods.cs:
```diff
@@ -295,5 +295,19 @@ public static void ReplaceWithGlobalConfirmMessage(this Control control)
+      /// Expand environment variables in a filename. Also expandes ~/ to %HOME% environment variable.
+      /// </summary>
+      /// <param name="path"></param>
+    public static String ExpandEnvironmentVariables(String filename)
```
To make this an extension method, write

    public static String ExpandEnvironmentVariables(this String filename)

dlech replied:
Then it can be used as 

      stringVar.ExpandEnvironmentVariables();

dlech commented:
If you make the method and actual extension method, you can write

     entrySettings.FileName.ExpandEnvironmentVariables();

works with code completion and everything

KuttKatrea commented:
Done, it's now a proper Extension method.

dlech commented:
Thanks! I cleaned up a few things and squashed the commits in f243409.
